plasmapy/particles/particle_class.py

```diff
@@ -2016,21 +2017,27 @@ class CustomParticle(AbstractPhysicalParticle):
 
     Parameters
     ----------
-    mass : ~astropy.units.Quantity, optional
-        The mass of the custom particle in units of mass.  Defaults to
-        |nan| kg.
+    *quantities : tuple of ~astropy.units.Quantity or `str`
+        The mass, electrical charge, and/or symbol of the custom
+        particle, provided as positional arguments. The mass and charge
+        may be provided in either order, but the symbol may only be the
+        final positional argument.
 
-    charge : ~astropy.units.Quantity or ~numbers.Real, optional
-        The electric charge of the custom particle.  If provided as a
-        `~astropy.units.Quantity`, then it must be in units of electric
-        charge. Defaults to |nan| C.
+    charge : ~astropy.units.Quantity or ~numbers.Real, |keyword-only|, optional
+        The electrical charge of the custom particle, if not provided in
+        ``quantities`` as a positional argument.
 
-    Z : ~numbers.Real, optional, |keyword-only|
-        The :term:`charge number`, which is equal to the ratio of the
-        charge to the elementary charge.
+    mass : ~astropy.units.Quantity, |keyword-only|, |keyword-only|, optional
+        The mass of the custom particle, if not provided in
+        ``quantities`` as a positional argument.
+
+    symbol : str, |keyword-only|, |keyword-only|, optional
+        The symbol to be assigned to the custom particle, if not
+        provided as the final positional argument.
 
-    symbol : str, optional
-        The symbol to be assigned to the custom particle.
+    Z : ~numbers.Real, optional, |keyword-only|
+        The |charge number|, which is equal to the ratio of the charge
+        to the elementary charge.
```

Comment:
I kind of dislike how this whole change actually made the user-facing complexity, as measured by the length of docstring necessary, higher! Nothing to change here, probably, just voicing my concern.

Reply:
Agreed — the downside of this change is the increased user-facing complexity.  My original plan was to deprecate the `mass` and `charge` keyword arguments, which I haven't done but am still considering.  The extra complexity comes from attempting to preserve backward compatibility where `mass` and `charge` can be used as keyword arguments too.